source/extensions/common/aws/utility.cc

```diff
@@ -25,19 +32,27 @@ Utility::canonicalizeHeaders(const Http::RequestHeaderMap& headers) {
         if (!entry.key().getStringView().empty() && entry.key().getStringView()[0] == ':') {
           return Http::HeaderMap::Iterate::Continue;
         }
+        // For S3, we restrict the headers even further to avoid  headers that might mutate
+        // when in transit (e.g.: x-forwarded-for, internal headers, etc).
+        if (*pair->second == "s3") {
+          const auto key = entry.key().getStringView();
+          if (key != Http::Headers::get().ContentType.get() && !absl::StartsWith(key, "x-amz-")) {
```

Comment:
I'm not sure I understand those two checks.
Shouldn't it just be if `if (key != Http::Headers::get().ForwardedProto)` then continue?

Reply:
you mean `Http::Headers::get().ForwardedFor()`? `x-forwarded-proto` is less likely to change, no? 

More generally though, I want to reduce the set of headers that are signed and per https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html the bare minimum are: `host`, `content-type` and `x-amzn-*`.

If we allow a broader range of headers to be signed, users might get errors because of headers that are dropped/removed after hence causing the signature check to failure.

Or am I missing something?